core/templating/datasource.go

```diff
@@ -53,3 +53,12 @@ func getData(source DataSource) (string, error) {
 	}
 	return csvData.String(), nil
 }
+
+func dataSourceExists(dataSources map[string]*DataSource, name string) bool {
```

Comment:
For more OOP pattern, this function should be on `TemplateDataSource`, as its interface should include, get / set / list / contain data source, etcs.

Reply:
Thanks, I will move it

Reply:
Done